packages/type-utils/src/TypeOrValueSpecifier.ts

```diff
@@ -0,0 +1,178 @@
+      },
+      required: ['from', 'name'],
+    },
+    {
+      type: 'object',
+      additionalProperties: false,
+      properties: {
+        from: {
+          type: 'string',
+          const: 'lib',
+        },
+        name: {
+          oneOf: [
+            {
+              type: 'string',
+            },
+            {
+              type: 'array',
+              minItems: 1,
+              uniqueItems: true,
+              items: {
+                type: 'string',
+              },
+            },
+          ],
+        },
+      },
+      required: ['from', 'name'],
+    },
+    {
+      type: 'object',
+      additionalProperties: false,
+      properties: {
+        from: {
+          type: 'string',
+          const: 'package',
+        },
+        name: {
+          oneOf: [
+            {
+              type: 'string',
+            },
+            {
+              type: 'array',
+              minItems: 1,
+              uniqueItems: true,
+              items: {
+                type: 'string',
+              },
+            },
+          ],
+        },
+        package: {
+          type: 'string',
+        },
+      },
+      required: ['from', 'name', 'package'],
+    },
+  ],
+};
+
+function specifierNameMatches(type: ts.Type, name: string | string[]): boolean {
+  if (typeof name === 'string') {
+    name = [name];
+  }
+  const symbol = type.getSymbol();
+  if (symbol === undefined) {
+    return false;
+  }
+  return name.some(item => item === symbol.escapedName);
+}
+
+function typeDeclaredInFile(
+  relativePath: string | undefined,
+  declarationFiles: ts.SourceFile[],
+  program: ts.Program,
+): boolean {
+  if (relativePath === undefined) {
+    return declarationFiles.some(declaration =>
+      declaration.fileName.startsWith(program.getCurrentDirectory()),
+    );
+  }
+  const absolutePath = path.join(program.getCurrentDirectory(), relativePath);
+  return declarationFiles.some(
+    declaration => declaration.fileName === absolutePath,
+  );
+}
+
+export function typeMatchesSpecifier(
+  type: ts.Type,
+  specifier: TypeOrValueSpecifier,
+  program: ts.Program,
+): boolean {
+  if (typeof specifier === 'string') {
+    return specifierNameMatches(type, specifier);
+  }
+  if (!specifierNameMatches(type, specifier.name)) {
+    return false;
+  }
+  const declarationFiles =
+    type
+      .getSymbol()
+      ?.getDeclarations()
+      ?.map(declaration => declaration.getSourceFile()) ?? [];
+  switch (specifier.from) {
+    case 'file':
+      return typeDeclaredInFile(specifier.path, declarationFiles, program);
+    case 'lib':
+      return declarationFiles.some(declaration =>
+        program.isSourceFileDefaultLibrary(declaration),
+      );
+    case 'package':
+      return declarationFiles.some(
+        declaration =>
+          declaration.fileName.includes(`node_modules/${specifier.package}/`) ||
+          declaration.fileName.includes(
+            `node_modules/@types/${specifier.package}/`,
```

Comment:
Ah I just remembered, package names with `@` in them have a `__` replacing the `@`. And there might be other replacements I'm not remembering - haven't looked deeply into it. 

https://github.com/DefinitelyTyped/DefinitelyTyped/tree/cc222f340546ff85b7c9271f4ecbd07b9d661c94/types/11ty__eleventy-img

@marekdedic I think this is fine to file as a followup issue for now - it should be quick to resolve. And I'd like to quit throwing up blockers to getting your PR merged 😄. Thoughts?

Reply:
Yes! I know that from the `@types/wordpress__data` packages! Great catch :D

Followup issue is fine by me ;)